pinot-spi/src/main/java/org/apache/pinot/spi/data/DateTimeFormatSpec.java

```diff
@@ -84,6 +127,23 @@ public DateTimeFormatSpec(int columnSize, String columnUnit, String columnTimeFo
     _patternSpec = new DateTimeFormatPatternSpec(columnTimeFormat);
   }
 
+  /**
+   * Constructs a dateTimeSpec format, given the components of a format
+   */
+  public DateTimeFormatSpec(String columnTimeFormat, String columnUnit, int columnSize) {
```

Comment:
Let's not add new constructors for this class. It should always be constructed from the format string

Reply:
If we don't add new constructors, then how can we know which delimiter to use?

Reply:
Actually, the tests are using the constructors. Writing the tests without the constructors would not be possible.

https://github.com/apache/pinot/blob/f8c728f8de4c393f4f768887351f244e0661b89d/pinot-common/src/test/java/org/apache/pinot/common/data/DateTimeFormatSpecTest.java#L403

Reply:
Ideally we shouldn't construct the format from the constructor that already have the parsed fields provided. But that is an existing problem and not introduced in this PR, and we can solve that separately.

Reply:
So should I remove the constructor and the tests altogether?

Reply:
Removed in the latest commit.